packages/commons-test/src/setupTestContainersVitestGlobal.ts

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+/* eslint-disable functional/immutable-data */
+/* eslint-disable functional/no-let */
+
+import { config as dotenv } from "dotenv-flow";
+import {
+  TEST_MINIO_PORT,
+  TEST_MONGO_DB_PORT,
+  TEST_POSTGRES_DB_PORT,
+  TestContainersConfig,
+  minioContainer,
+  mongoDBContainer,
+  postgreSQLContainer,
+} from "./containerTestUtils.js";
+import type {} from "vitest";
+
+declare module "vitest" {
+  export interface ProvidedContext {
+    config: TestContainersConfig;
+  }
+}
+
+export function setupTestContainersVitestGlobal() {
+  dotenv();
+  const config = TestContainersConfig.parse(process.env);
+  let teardown = false;
+
+  return async function ({
+    provide,
+  }: {
+    provide: (...args: unknown[]) => void;
+  }): Promise<() => Promise<void>> {
+    const startedPostgreSqlContainer = await postgreSQLContainer(
+      config
+    ).start();
+    const startedMongodbContainer = await mongoDBContainer(config).start();
+    const startedMinioContainer = await minioContainer(config).start();
+
+    config.eventStoreDbPort = startedPostgreSqlContainer.getMappedPort(
+      TEST_POSTGRES_DB_PORT
+    );
+    config.readModelDbPort =
+      startedMongodbContainer.getMappedPort(TEST_MONGO_DB_PORT);
+    config.s3ServerPort = startedMinioContainer.getMappedPort(TEST_MINIO_PORT);
+
+    provide("config", config);
```

Comment:
This `setupTestContainersVitestGlobal` will be run by vitest in a separate thread, we use this `provide` function to send `serializable` data to the thread in which the test will run. To "catch" this data we will use vitest's "inject" function.